### PR TITLE
updated PR template

### PR DESCRIPTION
**Pull request by dofinn**
### Which issue this PR addresses:

Fixes N/A

### What this PR does / why we need it:

Add light weight production operability considerations to PR. 

### Test plan for issue:

N/A 

### Is there any documentation that needs to be updated for this PR?

N/A

### CODE REVIEW

**Review by mociarain — COMMENTED** [review]
I don't disagree with this but I have 2 thoughts since we're looking at the template:

- Should we explicitly mention logging? 
- I worry that adding more sections will encourage larger single PR's that cover more and more.

What do you think?

**Comment by dofinn** [comment]
> I don't disagree with this but I have 2 thoughts since we're looking at the template:
> 
> * Should we explicitly mention logging?
> * I worry that adding more sections will encourage larger single PR's that cover more and more.
> 
> What do you think?

@mociarain I empathise with sizeable requirements for small improvements, but i hope people feel empowered to declare "NA" for sections. 

We can mention logging, but I prefer to leave the production operability open ended. What is it that you want to mention from logging? what if submitter has metrics/traces to utilise?

**Comment by mociarain** [comment]
Fair points. 

> We can mention logging, but I prefer to leave the production operability open ended. What is it that you want to mention from logging? what if submitter has metrics/traces to utilise?

I would suggest then:
"Does adequate telemetry, monitoring and documention exist to effectively operate your change?" -> either:

- Does adequate telemetry, logging, monitoring and documention exist to effectively operate your change?
- Does adequate production machinery (telemetry, logging etc.) exist to effectively operate your change?


> @mociarain I empathise with sizeable requirements for small improvements, but i hope people feel empowered to declare "NA" for sections.

That's true and I hope the same.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -33,3 +33,10 @@ How did you test that this PR works?
   docs.
 - If no, explain why (e.g. "tech debt cleanup, N/A").
 -->
+
+### How do you know this will function as expected in production? 
+
+<!--
+- Does adequate telemetry, monitoring and documention exist to effectively operate your change?
```
Comment (mociarain) [inline .github/PULL_REQUEST_TEMPLATE.md:40]:
```suggestion
- Does adequate telemetry, monitoring and documentation exist to effectively operate your change?
```

**Comment by dofinn** [comment]
> Fair points.
> 
> > We can mention logging, but I prefer to leave the production operability open ended. What is it that you want to mention from logging? what if submitter has metrics/traces to utilise?
> 
> I would suggest then: "Does adequate telemetry, monitoring and documention exist to effectively operate your change?" -> either:
> 
> * Does adequate telemetry, logging, monitoring and documention exist to effectively operate your change?
> * Does adequate production machinery (telemetry, logging etc.) exist to effectively operate your change?
> 
> > @mociarain I empathise with sizeable requirements for small improvements, but i hope people feel empowered to declare "NA" for sections.
> 
> That's true and I hope the same.

Logging is a form of telemetry. This statement is redundant. Looks like I need to submit PR against master branch anyway. CI can't run on forks. Will report back here and close this out.

**Comment by mociarain** [comment]
> Logging is a form of telemetry.

Is it? I wouldn't have grouped them in that way and a quick google points to them being separate but I'm sure we can find sources to back your view too.

Do we have a canonical definition at RH for this or something? Highly possible I missed this in all the onboarding :D

**Comment by dofinn** [comment]
https://github.com/Azure/ARO-RP/pull/3438 closing in favour of this

**Pull request closed without merging**